src/RewardsStreamerMP.sol

```diff
@@ -270,9 +277,15 @@ contract RewardsStreamerMP is
             vault.stakedBalance, vault.lockUntil, block.timestamp, vault.mpAccrued, vault.maxMP, amount
         );
         vault.stakedBalance -= amount;
-        vault.mpAccrued -= _deltaMpTotal;
         vault.maxMP -= _deltaMpMax;
         vault.rewardIndex = rewardIndex;
+        vault.mpAccrued -= _deltaMpTotal;
+
+        if (vault.mpAccrued < vault.mpStaked) {
+            vault.mpStaked = vault.mpAccrued;
+            totalMPStaked -= vault.mpStaked - vault.mpAccrued;
+        }
```

Comment:
Can you elaborate on these lines here?

As stated above, `mpAccrued` so far has been the "staked" or "settled" MPs.
Why are we performing this check here.

And why are we doing `vault.mpStasked = vault.mpAccrued` if `vault.mpAccrued < vault.mpStaked`?

I'm thinking of a scenario where alice has 400 `mpStaked` (initial plus bonus), her `accrued` (settled pending) is 20. So now we're setting her staked mps to `20` ?

I'm very likely missing something here.
Might be worth to clarify these lines with comments as well?

It also might make sense to revisit the naming of `accrued` vs `staked`

Reply:
if alices has 400 mpStaked and other 20 accrued, it means that accrued are 420.
accrued is the total (420), and some of those (400) are staked.

if shes unstakes something and we need to decrease here MPs by 10, then we will have:

* mpAccrued: 410
* mpStaked: 400

Instead, if she unstakes more and we need to decrease the MPs by 100, then `mpAccrued` will go from 420 to 320. This means she cannot have 400 MPs staked, cause she only has 320. So we decrease stakedMP to the maximum she can stakes, 320.

Reply:
Thank you for clarifying.
I'd still like to see a comment that clarifies this but we can do that in a follow-up PR.